helix-view/src/document.rs

```diff
@@ -1720,6 +1734,27 @@ impl Document {
         &self.selections
     }
 
+    pub(crate) fn get_view_data(&self, view_id: ViewId) -> Option<&ViewData> {
+        self.view_data.get(&view_id)
+    }
+    pub fn view_data(&self, view_id: ViewId) -> &ViewData {
+        self.view_data
+            .get(&view_id)
+            .expect("This should only be called after ensure_view_init")
+    }
+
+    pub fn view_data_mut(&mut self, view_id: ViewId) -> &mut ViewData {
+        self.view_data.entry(view_id).or_default()
+    }
+
+    pub fn view_offset(&self, view_id: ViewId) -> &ViewPosition {
```

Comment:
you don't need to return a reference here, the type is copy (its just a bunch of integers). That makes the api easier to use. its technically faster to return the pointer and deref at the callsite but llvm is very good at micro-optimizing that kind of thing we shouldn't worry about that here

Reply:
Done